pkg/operator/controller.go

```diff
@@ -584,12 +586,24 @@ func (c *Controller) syncDaemonSet(tuned *tunedv1.Tuned) error {
 		}
 	}
 
-	ds = ds.DeepCopy() // never update the objects from cache
-	ds.Spec = dsMf.Spec
-
 	if operatorReleaseVersion != operandReleaseVersion {
-		// Update the DaemonSet
 		klog.V(2).Infof("syncDaemonSet(): operatorReleaseVersion (%s) != operandReleaseVersion (%s), updating", operatorReleaseVersion, operandReleaseVersion)
+		update = true
+	}
+
+	// OCPBUGS-18480: sync the DaemonSet also when the operand image changes
+	operandImageCurrent := ds.Spec.Template.Spec.Containers[0].Image
+	operandImageWanted := os.Getenv("CLUSTER_NODE_TUNED_IMAGE")
+	if operandImageCurrent != operandImageWanted {
+		klog.V(2).Infof("syncDaemonSet(): operandImageCurrent (%s) != operandImageWanted (%s), updating", operandImageCurrent, operandImageWanted)
```

Comment:
just curious - can this not be controlled by defining the daemon set deployment manifest and specifying the image in the image-references file?

Reply:
Not really.  Then we'd pass the responsibility for creating the DaemonSet to CVO and you'd not be able to handle operator's ManagementState.